dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -564,14 +565,30 @@ public void delete(final Host host, final User user, final boolean respectFronte
         delete(host,user,respectFrontendRoles,false);
     }
 
-    public void delete(final Host host, final User user, final boolean respectFrontendRoles, boolean runAsSepareThread) {
-
+    public void delete(final Host host, final User deletingUser, final boolean respectFrontendRoles, boolean runAsSepareThread) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 31 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This method has 117 lines, which is greater than the 75 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS138)